io/directory.go

```diff
@@ -72,6 +76,12 @@ type Directory interface {
 type BasicDirectory struct {
 	node  *mdag.ProtoNode
 	dserv ipld.DAGService
+
+	// Internal variable used to cache the estimated size used for the
+	// HAMTShardingSize option. We maintain this value even if the
+	// HAMTShardingSize is off since potentially the option could be activated
+	// on the fly.
+	estimatedSize int
```

Comment:
The full documentation is in the exported `HAMTShardingSize` (referenced here) which is what the user has access to; this is just an internal cache. I'd like to retain this name (or something similar to this effect) to better represent *what* we want to achieve (estimation of directory size) rather than the *how* (traversing links).

Also not following the talk about a spec (even if figuratively) and a reference to an algorithm which I'm not sure what part of the code it is designating (is it the `addToEstimatedSize`/`removeFromEstimatedSize` functions?).

All this said, if this is a blocking change feel free to push any documentation you want or rename however you prefer. Not very opinionated on this and it won't change how this PR works anyway.

Reply:
I guess I just want something to indicate that this isn't just some random estimate that someone can just change, but that's not critical or merge blocking.